src/core/mac/mac_frame.hpp

```diff
@@ -45,6 +45,7 @@
 
 namespace ot {
 namespace Mac {
+class CslIe;
```

Comment:
What about moving the definition of `CslIe` ahead and avoid the forward declaration. Because no forward declaration is used in this source. And things like `TimeIe`, `ThreadIe` are defined ahead the `Frame` class.